### PR TITLE
Handling for data not found

### PR DESCRIPTION
**Pull request by wesleymonte**

### CODE REVIEW

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceLatencyProbe.java [hunk 1]
```diff
@@ -50,16 +49,13 @@ protected void populateMetadata(Map<String, String> metadata, Pair<String, Float
...
-        if(Objects.nonNull(lastMetricObservation)){
-            for(int i = 0; i < 3; i++){
-                if(list.get(i).getValue() == 0){
-                    list.set(i, lastMetricObservation.get(i));
-                }
+        List<Pair<String, Float>> list = new ArrayList<>();
+        for(Pair<String, Float> p : Arrays.asList(computeLatency, networkLatency, volumeLatency)){
+            //To avoid adding data when there are no audits to calculate
+            if(p.getValue() != 0){
+                list.add(p);
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceLatencyProbe.java:55]: Run the format tool on the files.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowResourceAvailabilityProbe.java [hunk 2]
```diff
@@ -48,35 +53,22 @@ protected void populateMetadata(Map<String, String> metadata, Pair<String, Float
...
+    private Pair<String, Float> getResourceAvailabilityValue(ResourceType type) throws Exception {
         LOGGER.debug("Getting audits from resource of type [" + type.getValue() + "]");
         Integer valueFailedAfterSuccessful = providerService
             .getAuditsFromResourceByState(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST, type,
                 lastTimestampAwake);
         Integer valueFulfilled = providerService
             .getAuditsFromResourceByState(OrderState.FULFILLED, type, lastTimestampAwake);
-        Float availabilityData = calculateAvailabilityData(valueFailedAfterSuccessful,
+        if(valueFailedAfterSuccessful == 0 && valueFulfilled == 0){
+            throw new Exception("Not found resource data to calculate resource availability.");
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowResourceAvailabilityProbe.java]: It doesn't smell good to throw `Exception`. You can throw a specific exception.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceSuccessRateProbe.java [hunk 3]
```diff
@@ -45,30 +50,18 @@ protected void populateMetadata(Map<String, String> metadata, Pair<String, Float
         metadata.put(RESOURCE_LABEL, p.getKey().toLowerCase());
     }
 
-    private Pair<String, Float> getResourceAvailabilityValue(ResourceType type) {
+    private Pair<String, Float> getResourceAvailabilityValue(ResourceType type) throws Exception {
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceSuccessRateProbe.java]: Same generic `Exception` thrown.

**Review by joivo — CHANGES_REQUESTED** [review]

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceLatencyProbe.java (hunk 1 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceLatencyProbe.java:55]: Done.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowResourceAvailabilityProbe.java (hunk 2 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowResourceAvailabilityProbe.java]: Created a new exception called OrdersStateChangeNotFoundException.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceSuccessRateProbe.java [hunk 4]
```diff
@@ -51,24 +57,12 @@ protected void populateMetadata(Map<String, String> metadata, Pair<String, Float
             .getAuditsFromResourceByState(OrderState.FAILED_ON_REQUEST, type, lastTimestampAwake);
         Integer valueOpen = providerService
             .getAuditsFromResourceByState(OrderState.OPEN, type, lastTimestampAwake);
-        Float availabilityData = calculateAvailabilityData(valueFailed, valueOpen);
+        if(valueFailed == 0 && valueOpen == 0){
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/fogbow/FogbowServiceSuccessRateProbe.java:60]: Run the format tool here too...

**Review by joivo — APPROVED** [review]

**Pull request merged**